specification/search/data-plane/Azure.Search/preview/2020-06-30/searchservice.json

```diff
@@ -6943,6 +6943,92 @@
       },
       "description": "Base type for skills."
     },
+    "CustomEntity": {
```

Comment:
2020-06-30 already GA'd. Unless this support already existed and just wasn't formalized in swagger, it should only be in 2020-06-30-Preview, which will change dates when closer to its GA.

Reply:
I actually have a question here given our case is a little bit involved.
The CustomEntityLookupSkill is publicly available. It's currently in our preview API and customers are using it right now.
The skill was added a while ago so it's not in the preview swagger.
Now our team wants to GA the feature so it should live in both the GA API and the preview API. 
Given the situation we are at, should I remove or keep the 2020-06-30 change?

Reply:
/cc @johanste @brjohnstmsft

Reply:
hi team, can I get any update on this?

Reply:
FYI @bleroy since this is about Skillsets

Reply:
This should be done consistently with our new story for skill versioning that @Careyjmac is building (provided timing allows it, but I'd still like to have her input for this specific case), but it should also be ruled by the general rules about breaking changes for existing APIs since this for all practical purposes seems like it's GA. I agree with @heaths' original comment that the GA version should be left alone and only the preview version should have the changes.

Reply:
Our skill story with regards to preview/GA is currently in a weird state.  Technically in the API on the backend, skills are version neutral and thus whenever we add a new skill or change an existing one, that change occurs across all API versions that support skillsets. We like to call skills "preview" when they are new by adding a preview disclaimer to the documentation and not adding SDK support for them until we decide to "GA" them but this is handwave-y because they are already technically exposed in GA API versions.

We understand this is probably not a very good practice and we are looking into ways to make the preview experience better for skills as part of our skill versioning design, which is scheduled to be implemented in Cobalt. For now, because we have decided to "GA" the CustomEntityLookupSkill via our handwave-y method of removing the preview disclaimer from the docs and adding it the the GA SDK swagger, Shuyang's change here makes sense.

Reply:
@Careyjmac please also consider the impact and versioning story of the SDK. To support this, the SDKs have to be regenerated, which first includes updating the commits to the swagger files. But since new APIs were added, strict semver rules dictate you now bump the minor version of the SDK packages. If skills (or any other models or methods) are added with some frequency, the package versions will iterate as frequently. This might become daunting for customers to know if they should upgrade when considering bug fixes will bump the patch version.

In such cases, bumping only the patch version for new models in the SDKs may make more sense, but is something the service and SDK teams should consider together.

Reply:
@Heaths of course.  We are still getting used to navigating this new world where the SDK team is so involved and all of the new breaking change requirements.  We are definitely considering the SDK implications with the new skill versioning discussions.  But since those discussions are ongoing and the work won't be done until Cobalt, I think taking a minor version bump for a change such as this one is acceptable for now.

Reply:
We're going to GA a minor version bump in a few weeks. @AlexGhiondea would adding a model require going through at least a single preview, or would this be minor enough (no pun intended) to include in our upcoming GA?

Reply:
@johanste how should this be handled going forward if new models are added to GAs? Isn't there roundtripping concerns here as well?

Reply:
Yes, there are some round-tripping concerns here. A client that knows nothing about a new skill type can easily end up stripping out properties it did not know about on update (assuming PUT does a replace rather than a merge-patch like update). I believe that I and @JeffreyRichter can be of assistance when it comes to determining (or at least reviewing) what a good versioning model for your service APIs should be.
Likewise, as mentioned above,

Reply:
+ @jennifermarsman since I know she was talking about revisting this design in the coming weeks.  We are hoping to finalize the design in Iron with the intent of implementing it in Cobalt, and we would definitely value any feedback from the SDK folks as we do so, so perhaps we should loop ya'll into our thread we have on the topic.